sw/device/lib/testing/test_framework/index.md

```diff
@@ -23,4 +30,19 @@ bool test_main() {
 }
 ```
 
-Check out the [rv\_timer smoke test](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/tests/rv_timer_smoketest.c) for an example chip-level test that uses the current test framework.
+Check out the [rv\_timer smoke test](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/tests/rv_timer_smoketest.c) for an example chip-level test that contains this boilerplate code.
+
+## Signaling the end of test and self-checking mechanism
+It is mandatory to invoke the target-agnostic API `test_status_set()` to explicitly signal the end of the test based on whether it passed or failed.
+When invoked, the API calls `abort()` at the end to stop the core from executing any further.
+Please see [`sw/device/lib/testing/test_framework/test_status.h`](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/test_framework/test_status.h) for documentation and usage.
```

Comment:
nit: Are we ok with this non-GitHub link? In most other in-tree docs, we are just referencing the relative paths (i.e. `sw/device/lib/testing/test_framework/test_status.h`).

Reply:
Some of these don't have an associated README.md/[_]index.md, but I fixed them to be github links.